src/include/duckdb/execution/operator/projection/physical_tableinout_function.hpp

```diff
@@ -46,6 +46,10 @@ class PhysicalTableInOutFunction : public PhysicalOperator {
 	//! Information for WITH ORDINALITY
 	optional_idx ordinality_idx;
 
+	bool SinkOrderDependent() const override {
+		return function.in_out_function_order_dependent;
```

Comment:
`PhysicalTableInOutFunction` is not a sink but an operator - so this will not have any effect. You want `OperatorOrder()`